cmd/http/listener.go

```diff
@@ -230,6 +232,7 @@ func (listener *httpListener) start() {
 					reqInfo.AppendTags("localAddr", bufconn.LocalAddr().String())
 					ctx := logger.SetReqInfo(context.Background(), reqInfo)
 					logger.LogIf(ctx, err)
+					bufconn.Write(badRequestMsg)
```

Comment:
Writing response just before closing is more appropriate.

Reply:
We don't need to write for network errors @balamurugana that is why this PR exists, we cannot write a response on an idle connection.

Reply:
IMO write and close is appropriate.  Either close or handle condition differently

Reply:
Not in this case, a network disconnection or a timeout doesn't mean that we should be writing back to the client which is meaningless. that is why code moved in situations only when it's not a network error we can write.